src/cpp/rtps/transport/TCPTransportInterface.cpp

```diff
@@ -1744,14 +1746,11 @@ void TCPTransportInterface::SocketConnected(
 
         if (channel)
         {
-            if (!error)
+            if (!error && TCPChannelResource::eConnectionStatus::eConnecting == channel->connection_status())
```

Comment:
I changed this condition to something I believe makes more sense, but please double check I am not missing a transition covered by the prior condition.